#!/usr/bin/env python
PACKAGE = "slalom"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ouster_frame", str_t, 0, "Melyik frame-en levo oustert hasznaljuk","right_os1/os1_sensor")
gen.add("topic_name", str_t,0,"Melyik topicra iratkozunk_fel","velodyne_points")
gen.add("dbscan_eps",    double_t,    0, "A DBSCAN eps parametere", 0.5,  0.0,2.0 )
gen.add("dbscan_min_samples",    int_t,    0, "A DBSCAN minimum pontok szama", 15,  0,50 )
gen.add("max_radius",    double_t,    0, "A pontokra illesztett kor max sugara", 1.0,  0.0,5.0 )
gen.add("min_radius",    double_t,    0, "A pontokra illesztett kor min sugara", 0.3,  0.0,5.0 )
gen.add("max_distance",    double_t,    0, "A ket henger kozotti tavolsag max", 3.0,  0.0,5.0 )
gen.add("min_distance",    double_t,    0, "A ket henger kozotti tavolsag min", 2.0,  0.0,5.0 )
gen.add("min_x",    double_t,    0, "A pontfelho minimalis x iranyu tavolsag", 1.5,  -30.0, 30.0)
gen.add("max_x",    double_t,    0, "A pontfelho maximalis x iranyu tavolsag", 30.0,  -30.0, 35.0)
gen.add("min_y",    double_t,    0, "A pontfelho minimalis y iranyu tavolsag", -7.0,  -30.0, 30.0)
gen.add("max_y",    double_t,    0, "A pontfelho maximalis y iranyu tavolsag", 7.0,  -30.0, 30.0)
gen.add("barrier_z",    double_t,    0, "Az oldal terelo elem magassaga ", 0.5,  0.0, 2.0)

exit(gen.generate(PACKAGE, "slalom", "Slalom"))